src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,82 @@ class C1<T>
+            string source = @"
+unsafe class C
+{
+    delegate*<out string, ref readonly int> M(delegate* cdecl<void> param) => throw null;
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics();
+
+            var sourceModule = comp.SourceModule;
+            var sourceAssembly = (SourceAssemblySymbol)sourceModule.ContainingAssembly;
+            var sourceNamespace = sourceModule.GlobalNamespace;
+
+            var retargetingAssembly = new RetargetingAssemblySymbol(sourceAssembly, isLinked: false);
+            retargetingAssembly.SetCorLibrary(sourceAssembly.CorLibrary);
+            var retargetingModule = retargetingAssembly.Modules[0];
+            var retargetingNamespace = retargetingModule.GlobalNamespace;
+
+            RetargetingSymbolChecker.CheckSymbols(sourceNamespace.GetMember<NamedTypeSymbol>("C"), retargetingNamespace.GetMember<NamedTypeSymbol>("C"));
+        }
+
+        [Fact]
+        public void RetargetingFunctionPointerType_02()
+        {
+            var source = @"
+public unsafe class C
+{
+    public delegate* cdecl<out string, int> M() => throw null;
+}
+";
+
+            CSharpCompilation c1 = CreateCompilation(source, targetFramework: TargetFramework.Mscorlib40, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularPreview);
+
+            var c1Assembly = c1.Assembly;
+
+            var r1 = c1.ToMetadataReference();
+            CSharpCompilation c2 = CreateCompilation("", references: new[] { r1 }, targetFramework: TargetFramework.Mscorlib45);
+            var c1AsmRef = c2.GetReferencedAssemblySymbol(r1);
+
+            var c2CorLib = c2.Assembly.CorLibrary;
+
+            Assert.NotSame(c1Assembly, c1AsmRef);
+            Assert.NotSame(c2.Assembly.CorLibrary, c1.Assembly.CorLibrary);
+
+            var ptr_c1 = getFuncPtr(c1);
+            var ptr_c2 = getFuncPtr(c2);
+
+            Assert.NotSame(ptr_c1, ptr_c2);
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptr_c1);
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptr_c2);
+
+            Assert.NotSame(c2CorLib, ptr_c1.Signature.ReturnType.ContainingAssembly);
+            Assert.Same(c2CorLib, ptr_c2.Signature.ReturnType.ContainingAssembly);
+
+            Assert.Equal(ptr_c1.Signature.CallingConvention, ptr_c2.Signature.CallingConvention);
+
+            var param_c1 = ptr_c1.Signature.Parameters.Single();
+            var param_c2 = ptr_c2.Signature.Parameters.Single();
+
+            Assert.NotSame(c2CorLib, param_c1.Type.ContainingAssembly);
+            Assert.Same(c2CorLib, param_c2.Type.ContainingAssembly);
+
+            Assert.NotSame(c2CorLib, ((CSharpCustomModifier)param_c1.RefCustomModifiers.Single()).ModifierSymbol.ContainingAssembly);
+            Assert.Same(c2CorLib, ((CSharpCustomModifier)param_c2.RefCustomModifiers.Single()).ModifierSymbol.ContainingAssembly);
```

Comment:
>Assert.Same(c2CorLib, ((CSharpCustomModifier)param_c2.RefCustomModifiers.Single()).ModifierSymbol.ContainingAssembly); [](start = 12, length = 118)

It would be good to test:
- retargeting of ref modifiers on the signature
- retargeting of modifiers on return types and parameter types. However, the way to get them there would be through inheritance during overriding and I am not sure if that piece is implemented yet.